processor/transformprocessor/internal/common/functions.go

```diff
@@ -70,6 +71,32 @@ func keepKeys(target GetSetter, keys []string) ExprFunc {
 	}
 }
 
+func truncateAll(target GetSetter, limit int64) ExprFunc {
```

Comment:
Limit should be an unsigned integer (or at least passing negative integers should make the transform processor fail validation)

Reply:
The parser currently always returns an `int64` because it doesn't know semantics yet, which is what's passed here. So indeed, validation here would be good - currently the factories return `ExprFunc` but it seems they should actually be `ExprFunc, error`

Reply:
If this is not done anywhere else, it's fine for me to leave this for a different PR

Reply:
I'd not in place yet, but I see how it could be done.  I will open a new issue to add that functionality to the framework.  In the meantime I added a check in the function itself so that limits < 0 do nothing.  I also added a new test.